File: Persistencia/Scripts/micro_dao/comentario.py
from micro_dao.objeto_modelo import ObjetoModelo

class Comentario(ObjetoModelo):

    _id: int
    _comentario: str
    _fk_avaliacao: int

    def __init__(self, comentario: str, fk_avaliacao: int):
        self.set_comentario(comentario)
        self.set_fk_avaliacao(fk_avaliacao)

    def get_id(self) -> int:
        return self._id

    def set_id(self, id: int):
        self._id = id
        return self

    def get_comentario(self) -> str:
        return self._comentario

    def set_comentario(self, comentario: str):
        self._comentario = comentario
        return self

    def set_fk_avaliacao(self, fk_avaliacao: int):
        self._fk_avaliacao = fk_avaliacao
        return self

    def get_fk_avaliacao(self) -> int:
        return self._fk_avaliacao

File: Projeto_Java/src/Dominio/Entidades/Comentario.java
package Dominio.Entidades;

import javax.persistence.*;

@Entity
@Table(name = "comentario")
public class Comentario {

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "comentario", length = 500)
    private String comentario;

    @Column(name = "fk_avaliacao")
    private int fkAvaliacao;


    public Comentario() { }

    public Comentario(String comentario, int fkAvalicao) {
        this.setComentario(comentario);
        this.setFkAvalicao(fkAvalicao);
    }

    public Comentario(int id, String comentario, int fkAvalicao) {
        this.setId(id);
        this.setComentario(comentario);
        this.setFkAvalicao(fkAvalicao);
    }


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getComentario() {
        return comentario;
    }

    public void setComentario(String comentario) {
        this.comentario = comentario;
    }

    public int getFkAvalicao() {
        return fkAvaliacao;
    }

    public void setFkAvalicao(int fkAvalicao) {
        this.fkAvaliacao = fkAvalicao;
    }

    @Override
    public String toString() {
        return String.format(
                "ID:\t\t\t\t%d;\nComentário:\t\t'%s';\nFK_avaliacao:\t%d;",
                this.getId(), this.getComentario(), this.getFkAvalicao());
    }
}
